src/compas/datastructures/mesh/core/mesh.py

```diff
@@ -1286,32 +1286,36 @@ def vertices_on_boundary(self, ordered=False):
         >>>
 
         """
-        vertices_set = set()
-        for key, nbrs in iter(self.halfedge.items()):
-            for nbr, face in iter(nbrs.items()):
-                if face is None:
-                    vertices_set.add(key)
-                    vertices_set.add(nbr)
-        vertices = list(vertices_set)
-
-        if not vertices:
-            return vertices
-        if not ordered:
-            return vertices
-
-        key = sorted([(key, self.vertex_coordinates(key)) for key in vertices], key=lambda x: (x[1][1], x[1][0]))[0][0]
-
-        vertices = []
-        start = key
-        while 1:
-            for nbr, fkey in iter(self.halfedge[key].items()):
-                if fkey is None:
-                    vertices.append(nbr)
-                    key = nbr
-                    break
-            if key == start:
-                break
-        return vertices
+        # vertices_set = set()
+        # for key, nbrs in iter(self.halfedge.items()):
+        #     for nbr, face in iter(nbrs.items()):
+        #         if face is None:
+        #             vertices_set.add(key)
+        #             vertices_set.add(nbr)
+        # vertices = list(vertices_set)
+
+        # if not vertices:
+        #     return vertices
+        # if not ordered:
+        #     return vertices
+
+        # key = sorted([(key, self.vertex_coordinates(key)) for key in vertices], key=lambda x: (x[1][1], x[1][0]))[0][0]
+
+        # vertices = []
+        # start = key
+        # while True:
+        #     for nbr, fkey in iter(self.halfedge[key].items()):
+        #         if fkey is None:
+        #             vertices.append(nbr)
+        #             key = nbr
+        #             break
+        #     if key == start:
+        #         break
+        # return vertices
+        boundaries = self.vertices_on_boundaries()
+        if boundaries:
+            return boundaries[0]
+        return []
```

Comment:
maybe debatable opinion but here I feel the one-liner actually improves readability:

```suggestion
        return boundaries[0] if boundaries else []
```